libraries/chain/include/eosio/chain/log_data_base.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+#include <boost/iostreams/device/mapped_file.hpp>
+
+namespace eosio {
+namespace chain {
+
+template <typename T>
+T read_buffer(const char* buf) {
+   T result;
+   memcpy(&result, buf, sizeof(T));
+   return result;
+}
+
+template <typename Derived>
+class log_data_base {
+ protected:
+   boost::iostreams::mapped_file file;
```

Comment:
Using a mapped file to supply log data makes it significantly harder to recover from IO errors. As is, should a log entry fail to read, it will SIGBUS the process I believe -- effectively SIGKILLing it. That seems rather extreme failure mode to me, particularly for something like the block log where it's easy to imagine such a failure could nominally be swallowed.

Reply:
the code has been converted to regular file IO instead memory mapped file IO.